### PR TITLE
add error validation to executors for executor_queue_name env var

### PR DESCRIPTION
**Pull request by DaedalusG**
Resolves https://github.com/sourcegraph/sourcegraph/issues/41089

## Test plan

Tested Locally with `env EXECUTOR_QUEUE_NAME=prod-codeintel sg start codeintel`
Screen Shot 2022-09-07 at 10 24 41 PM

### CODE REVIEW

**Review by chrismwendt — APPROVED** [review]
🚀

**Comment by DaedalusG** [comment]
First ever pull request against `sourcegraph/sourcegraph` (_for code_) 🎉

File: enterprise/cmd/executor/config.go [hunk 1]
```diff
@@ -84,6 +84,9 @@ func (c *Config) Validate() error {
 		// Required by Firecracker: The vCPU number is invalid! The vCPU number can only be 1 or an even number when hyperthreading is enabled
 		c.AddError(errors.Newf("EXECUTOR_JOB_NUM_CPUS must be 1 or an even number"))
 	}
+	if c.QueueName != "batches" && c.QueueName != "codeintel" {
```
Comment (eseliger) [inline enterprise/cmd/executor/config.go:87]: Can you please add a note to [the comment here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/cmd/frontend/internal/executorqueue/init.go?L28-30) that it's required to change this code as well when adding a new queue.

**Review by eseliger — APPROVED** [review]
Awesome 🔥 🚀 

The Go 🏅 is yours :)

**Pull request merged**